OMCompiler/Compiler/runtime/System_omc.c

```diff
@@ -758,27 +758,35 @@ extern const char* System_snprintff(const char *fmt, int len, double d)
     MMC_THROW();
   }
   buf = ModelicaAllocateString(len);
-  if (snprintf(buf,len,fmt,d) >= len) {
+
+  int str_len = snprintf(buf, len, fmt, d);
+
+  if (str_len < 0) {
+    MMC_THROW();
+  }
+
+  if (str_len > len) {
```

Comment:
And error message `System_snprintff: Buffer too small.` (or something like this)